stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -563,16 +608,127 @@ impl NakamotoBlock {
             .flatten()
     }
 
+    /// Determine if this is a well-formed tenure-extend block.
+    /// * It has exactly one TenureChange, and it does _not_ require a sortiton (it's `cause` is
+    /// `Extended`)
+    /// * Its consensus hash and previous consensus hash values point to this block.
+    /// * There is no coinbase
+    /// * There are no other TenureChange transactions
+    ///
+    /// Returns Some(true) if the above are true
+    /// Returns Some(false) if at least one of the above is false
+    /// Returns None if this block is not a tenure-extend block
```

Comment:
These `is_wellformed` functions seem like pretty un-ergonomic return types. `Some(false)` indicates a block is invalid -- which should almost always manifest in an InvalidBlock. It seems more logical to have something like:

```
Result<bool, ChainstateError>
```

Where the `Ok(true)` indicates that the block is an extend block, `Ok(false)` indicates that it is not, and `Err` is an invalid block.

Reply:
Sure; will use `Result<bool, ()>`